### PR TITLE
Custom directory flag for augeas lenses.

### PR DESCRIPTION
**Pull request by spasam**
This provides ability for users to specify their own augeas lenses directory.

On OS X, docker top API is not honoring 'ps_args'. Added check to make results
have correct number of columns.

On Ubuntu 16.04, I am unable to 'make deps' without libyaml. pip install of
requirements.txt is failing to link. Complaining about missing libyaml.

### CODE REVIEW

File: osquery/tables/system/posix/augeas.cpp [hunk 1]
```diff
@@ -15,13 +15,25 @@
...
+namespace osquery {
+
+/**
+ * @brief Augeas lenses path.
+ *
+ * Directory that contains augeus lenses.
+ */
 #ifdef __APPLE__
-#define LENSES_PATH "/private/var/osquery/lenses"
+FLAG(string,
```
Comment (theopolis) [inline osquery/tables/system/posix/augeas.cpp]: Can you bring the `FLAG` macro out of the `ifdef`, and just have 1 flag definition with the variable `LENSES_PATH` macro?

File: tools/provision.sh [hunk 2]
```diff
@@ -63,6 +63,7 @@ function platform_linux_main() {
   brew_tool libidn
   brew_tool libedit
   brew_tool libtool
+  brew_tool libyaml
```
Comment (theopolis) [inline tools/provision.sh]: por qué?

File: osquery/tables/applications/posix/docker.cpp [hunk 3]
```diff
@@ -543,6 +543,9 @@ QueryData genContainerProcesses(QueryContext& context) {
         BOOST_FOREACH (const auto& v, processes.second) {
           vector.push_back(v.second.data());
         }
+        if (vector.size() != 21) {
```
Comment (theopolis) [inline osquery/tables/applications/posix/docker.cpp]: Good catch! Maybe this should be a separate commit?

Comment (spasam) [inline osquery/tables/applications/posix/docker.cpp]: Sure. Will create another pull request.

File: tools/provision.sh (hunk 2 above)
Comment (spasam) [inline tools/provision.sh]: On Ubuntu 16.04, I am unable to 'make deps' without libyaml. pip install of requirements.txt is failing to link. Complaining about missing libyaml. Not sure why no one else is seeing this! I can remove this and work-around!

File: osquery/tables/system/posix/augeas.cpp (hunk 1 above)
Comment (spasam) [inline osquery/tables/system/posix/augeas.cpp]: :laughing: I tried that first. Got this warning:

osquery/osquery/tables/system/posix/augeas.cpp:27:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
#ifdef __APPLE__
 ^
osquery/osquery/tables/system/posix/augeas.cpp:31:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
#endif
 ^

Comment (spasam) [inline osquery/tables/system/posix/augeas.cpp]: Ok. split the commit into separate pull requests. Let me know if there is a way to work-around above warnings. Thanks

**Comment by theopolis** [comment]
ok to test

File: osquery/tables/system/posix/augeas.cpp (hunk 1 above)
Comment (theopolis) [inline osquery/tables/system/posix/augeas.cpp]: Eek, wacky, well then, let's go with this for now.

**Comment by theopolis** [comment]
Ah, sorry I didn't catch this before @spasam, but do you mind adding this `--augeas_lenses` flag to the `cli-flags.md` wiki page?
